balta/src/main/scala/za/co/absa/db/balta/classes/QueryResultRow.scala

```diff
@@ -16,83 +16,154 @@
 
 package za.co.absa.db.balta.classes
 
-import org.postgresql.util.PGobject
-import za.co.absa.db.balta.classes.simple.JsonBString
+import QueryResultRow._
 
-import java.sql.{Date, ResultSet, Time}
-import java.time.{Instant, OffsetDateTime}
+import java.sql
+import java.sql.{Date, ResultSet, ResultSetMetaData, Time, Types}
+import java.time.{Instant, LocalDateTime, OffsetDateTime, OffsetTime}
 import java.util.UUID
 
 /**
  * This is a row of a query result. It allows to safely extract values from the row by column name.
  *
- * @param resultSet - the JDBC result of a query
+ * @param rowNumber   - the number of the row in the result set
+ * @param fields      - the values of the row
+ * @param columnNames - the names of the columns
  */
-class QueryResultRow private[classes](val resultSet: ResultSet) extends AnyVal {
-  // this is not stable as resultSet mutates, but good enough for now
-  private def safe[T](fnc: => T): Option[T] = {
-    val result = fnc
-    if (resultSet.wasNull()) {
-      None
-    } else {
-      Some(result)
-    }
-  }
+class QueryResultRow private[classes](val rowNumber: Int,
```

Comment:
`rowNumber` is never used. Is some functionality yet to be added?

Reply:
_IntelliJ_ is to eager on this reporting IMHO.
This is a library. Right here in the library, the property is not used. But in tests it actually can be used.

Reply:
hmm, the reasoning of having a parameter supplied by caller to callee and having such parameter never used is just odd - caller knows about this parameter / value already anyway since he must supply it

I don't understand the reasoning about tests & I don't think that it makes sense to have something for tests only for benefit of tests 🤔

Reply:
Well the `rowNumber` is provided by `QueryResult` when it creates the `QueryResultRow`, it's not available to the user of the iterator (user of _Balta_ library) directly, but only via the `QueryResultRow` the iterator `QueryResult` returns.

Reply:
Hmm, that sounds okay